airbyte-integrations/connectors/source-google-drive/source_google_drive/stream_reader.py

```diff
@@ -0,0 +1,163 @@
+        seen: Set[str] = set()
+        while len(folder_id_queue) > 0:
+            (path, folder_id) = folder_id_queue.pop()
+            # fetch all files in this folder (1000 is the max page size)
+            request = service.files().list(
+                q=f"'{folder_id}' in parents", pageSize=1000, fields="nextPageToken, files(id, name, modifiedTime, mimeType)"
+            )
+            while True:
+                results = request.execute()
+                new_files = results.get("files", [])
+                for new_file in new_files:
+                    # It's possible files and folders are linked up multiple times, this prevents us from getting stuck in a loop
+                    if new_file["id"] in seen:
+                        continue
+                    seen.add(new_file["id"])
+                    file_name = path + new_file["name"]
+                    if new_file["mimeType"] == FOLDER_MIME_TYPE:
+                        folder_id_queue.append((f"{file_name}/", new_file["id"]))
+                        continue
+                    last_modified = datetime.strptime(new_file["modifiedTime"], "%Y-%m-%dT%H:%M:%S.%fZ")
+                    remote_file = GoogleDriveRemoteFile(
+                        uri=file_name, last_modified=last_modified, id=new_file["id"], mimeType=new_file["mimeType"]
+                    )
+                    if self.file_matches_globs(remote_file, globs):
+                        yield remote_file
+                request = service.files().list_next(request, results)
+                if request is None:
+                    break
+
+    def _get_folder_id(self, url):
+        # Regular expression pattern to check the URL structure and extract the ID
+        pattern = r"^https://drive\.google\.com/drive/folders/([a-zA-Z0-9_-]+)$"
+
+        # Find the pattern in the URL
+        match = re.search(pattern, url)
+
+        if match:
+            # The matched group is the ID
+            drive_id = match.group(1)
+            return drive_id
+        else:
+            # If no match is found
+            raise ValueError(f"Could not extract folder ID from {url}")
+
+    def open_file(self, file: GoogleDriveRemoteFile, mode: FileReadMode, encoding: Optional[str], logger: logging.Logger) -> IOBase:
+        if file.mimeType.startswith(EXPORTABLE_DOCUMENTS_MIME_TYPE_PREFIX):
+            if mode == FileReadMode.READ:
+                raise ValueError("Cannot read Google Docs/Sheets/Presentations and so on as text. Please set the format to PDF")
+            request = self.google_drive_service.files().export_media(fileId=file.id, mimeType=self._get_export_mime_type(file))
+        else:
+            request = self.google_drive_service.files().get_media(fileId=file.id)
+        handle = io.BytesIO()
+        downloader = MediaIoBaseDownload(handle, request)
+        done = False
+        while done is False:
+            status, done = downloader.next_chunk()
```

Comment:
Nit: since we're not using `status` this can be
```suggestion
            _, done = downloader.next_chunk()
```